### PR TITLE
rename fields

### PR DESCRIPTION
**Pull request by ehwenk**
taxonomic_reference -> taxonomic_dataset
cleaned_name -> aligned_name (and variants thereof)

### CODE REVIEW

File: tests/testthat/examples/Test_2023_2/output/taxa.csv [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
-Psychotria sp. Utchee Creek (H.Flecker NQNC5313),,,,,,,,,
-Quassia baileyana,,,,,,,,,
-Rhodomyrtus trineura,,,,,,,,,
-Rockinghamia angustifolia,,,,,,,,,
-Syzygium gustavioides,,,,,,,,,
-Syzygium sayeri,,,,,,,,,
-Trema aspera,,,,,,,,,
+aligned_name,taxon_name,taxonomic_dataset,taxon_rank,taxonomic_status,genus,binomial,trinomial,family,taxon_id,taxon_distribution
+Acacia celsa,Acacia celsa,APC,species,accepted,Acacia,Acacia celsa,,Fabaceae,https://id.biodiversity.org.au/taxon/apni/51436506,Qld
+,Acmena graveolens,,,,,,,,,
```
Comment (dfalster) [inline tests/testthat/examples/Test_2023_2/output/taxa.csv:3]: did you mean to add a blank cell at the start of each row?

File: tests/testthat/examples/Test_2023_3/output/taxa.csv [hunk 2]
```diff
@@ -1,186 +1,186 @@
...
-Swainsona cadellii,,,,,,,,,
-Teucrium puberulum,,,,,,,,,
-Thelymitra kangaloonica,,,,,,,,,
-Thesium australe,,,,,,,,,
-Trifolium dubium,,,,,,,,,
-Veronica catenata,,,,,,,,,
-Veronica derwentiana,,,,,,,,,
-Wahlenbergia luteola,,,,,,,,,
+aligned_name,taxon_name,taxonomic_dataset,taxon_rank,taxonomic_status,genus,binomial,trinomial,family,taxon_id,taxon_distribution
+,Abildgaardia ovata,,,,,,,,,
```
Comment (dfalster) [inline tests/testthat/examples/Test_2023_3/output/taxa.csv:2]: did you mean to add a blank cell at the start of each row?

File: R/process.R [hunk 3]
```diff
@@ -191,16 +191,18 @@ dataset_process <- function(filename_data_raw,
   # Record methods
   methods <- process_format_methods(metadata, dataset_id, sources, contributors)
 
+  excluded_taxa <- metadata$exclude_observations %>% util_list_to_df2()
```
Comment (dfalster) [inline R/process.R:194]: Is this meant to be added? Seems unrelated to everything else in this PR

**Review by dfalster — APPROVED** [review]

File: R/process.R (hunk 3 above)
Comment (ehwenk) [reply R/process.R:194]: No, this wasn't intended - I had a great tangling of branches, trying to do two things at once, and this line slipped through. Meanwhile a different one was missed.

File: tests/testthat/examples/Test_2023_3/output/taxa.csv (hunk 2 above)
Comment (ehwenk) [reply tests/testthat/examples/Test_2023_3/output/taxa.csv:2]: It was intentional, but open to discussion. I added `aligned_name` to the taxa table, but went back and forth several times as to whether this was a good/bad idea. Now leaning back to removing it.

File: tests/testthat/examples/Test_2023_2/output/taxa.csv (hunk 1 above)
Comment (ehwenk) [reply tests/testthat/examples/Test_2023_2/output/taxa.csv]: removed them again - had added `aligned_name`

**Comment by ehwenk** [comment]
Closes issue #98
